From 84322bcb96d5c1bbb5407705d8847a2b8b997615 Mon Sep 17 00:00:00 2001
From: Thomas Loimer <thomas.loimer@tuwien.ac.at>
Date: Tue, 5 Sep 2023 09:53:42 +0200
Subject: [PATCH 2/9] Do not close stdin, ticket #156

Under Darwin and FreeBSD, test 17 "read compressed image files" would
fail. The test converts a fig file which contains an embedded gif image
to eps. The gif image is given as "line.gif" in the fig file, but exists
as "line.gif.gz" on disk. Fig2dev effectively invokes a pipe
`popen("gunzip -c line.gif.gz", "r")` and feeds the uncompressed image
to another pipe
`popen("giftopnm -quiet | ppmtopcx -quit >'/tmp/tmp.pcx'", "w")`. The
error message was "giftopnm: ...Error reading file. errno=9 (Bad file
descriptor)". The error occured if the fig file was fed to fig2dev via
stdin (cat img.fig | fig2dev -L eps >img.eps), not if it was read from
disk (fig2dev -L eps img.fig img.eps).
---
 fig2dev/read.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/fig2dev/read.c b/fig2dev/read.c
index cb8f778..aa68084 100644
--- a/fig2dev/read.c
+++ b/fig2dev/read.c
@@ -152,7 +152,8 @@ readfp_fig(FILE *fp, F_compound *obj)
 		status = read_objects(fp, obj);
 	else
 		status = read_1_3_objects(fp, obj);
-	(void)fclose(fp);
+	if (fp != stdin)
+		(void)fclose(fp);
 	return status;
 }
 
-- 
2.34.1

